core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -504,6 +518,7 @@ private[master] class Master(
   }
 
   private def completeRecovery() {
+    // TODO Why synchronized
```

Comment:
This was due to an earlier state in the code when this method could be invoked from a different thread. It can be safely removed now.